lib/instance_billing_flavor_check/utils.py

```diff
@@ -240,25 +240,33 @@ def make_request(rmt_ip_addr, metadata, identifier):
             message = 'Request error:{}'.format(err)
         except Exception as err:
             message = 'Unexpected error: {}'.format(err)
-    
+
         if message:
-            logger.warning(
-                'Attempt {}: failed: {}'.format(retry_count, message)
-            )
-            if 'Timeout' in message:
+            if 'Timeout' in message or 'Connecting' in message:
+                logger.warning(
+                    'Attempt {}: failed: {}'.format(retry_count, message)
+                )
                 retry_count += 1
+                if retry_count == 4:
+                    return
```

Comment:
This is not necessary we will exit the while loop when we hit 4

Reply:
you don't like this check but without this check we introduce an infinite loop

if the error is not `timeout` or `connecting` the code is in a `while` loop until `retry_count` is 4,
which will never be because the retry_count gets updated inside the timeout or connecting error,
so I think we should keep this check on this PR

if we used a `for` loop with range(), we would not have an infinite loop, we would have (3) unnecessary
iterations if the error is not `timeout` or `connecting` but the loop would exit

Reply:
fixed in [41ecab0](https://github.com/SUSE-Enceladus/instance-billing-flavor-check/pull/24/commits/41ecab0a9b3c4df3327ea69be8b30f46b0d6f600)